cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -0,0 +1,546 @@
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+/**
+ * <p>Assembly operator for a "Kafka Cluster Rebalance" assembly, which:</p>
+ * <ul>
+ *     <li>Interact with Cruise Control REST API</li>
+ * </ul>
+ */
+public class KafkaClusterRebalanceAssemblyOperator
+        extends AbstractOperator<KafkaClusterRebalance, AbstractWatchableResourceOperator<KubernetesClient, KafkaClusterRebalance, KafkaClusterRebalanceList, DoneableKafkaClusterRebalance, Resource<KafkaClusterRebalance, DoneableKafkaClusterRebalance>>> {
+
+    private static final Logger log = LogManager.getLogger(KafkaClusterRebalanceAssemblyOperator.class.getName());
+
+    public static final String ANNO_STRIMZI_IO_REBALANCE = Annotations.STRIMZI_DOMAIN + "/rebalance";
+    private static final long REBALANCE_TASK_STATUS_TIMER_MS = TimeUnit.SECONDS.toMillis(5);
+
+    private final CrdOperator<KubernetesClient, KafkaClusterRebalance, KafkaClusterRebalanceList, DoneableKafkaClusterRebalance> clusterRebalanceOperator;
+    private final CrdOperator<KubernetesClient, Kafka, KafkaList, DoneableKafka> kafkaOperator;
+    private final PlatformFeaturesAvailability pfa;
+    private final Function<Vertx, CruiseControlApi> cruiseControlClientProvider;
+
+    /**
+     * @param vertx The Vertx instance
+     * @param pfa Platform features availability properties
+     * @param supplier Supplies the operators for different resources
+     */
+    public KafkaClusterRebalanceAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
+                                                 ResourceOperatorSupplier supplier) {
+        this(vertx, pfa, supplier, v -> new CruiseControlApiMockImpl(vertx));
+    }
+
+    public KafkaClusterRebalanceAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
+                                                 ResourceOperatorSupplier supplier,
+                                                 Function<Vertx, CruiseControlApi> cruiseControlClientProvider) {
+        super(vertx, KafkaClusterRebalance.RESOURCE_KIND, supplier.kafkaClusterRebalanceOperator);
+        this.pfa = pfa;
+        this.clusterRebalanceOperator = supplier.kafkaClusterRebalanceOperator;
+        this.kafkaOperator = supplier.kafkaOperator;
+        this.cruiseControlClientProvider = cruiseControlClientProvider;
+    }
+
+    /**
+     * Create a watch on {@code KafkaClusterRebalance} in the given {@code namespace}.
+     *
+     * @param watchNamespaceOrWildcard The namespace to watch.
+     * @return A future which completes when the watch has been set up.
+     */
+    public Future<Void> createClusterRebalanceWatch(String watchNamespaceOrWildcard) {
+
+        return Util.async(this.vertx, () -> {
+            this.clusterRebalanceOperator.watch(watchNamespaceOrWildcard, new Watcher<KafkaClusterRebalance>() {
+                @Override
+                public void eventReceived(Action action, KafkaClusterRebalance kafkaClusterRebalance) {
```

Comment:
Well it doesn't raise any error and goes smoothly because:

* on DELETED, the `kafkaClusterRebalance` instance will not be `null` anyway
* when `reconcileClusterRebalance` is called and if action is DELETED, `null` is passed to it